examples/expectations/regex_based_column_map_expectation_template.py

```diff
@@ -0,0 +1,78 @@
+"""
+This is a template for creating custom ColumnMapExpectations.
+For detailed instructions on how to use it, please see:
+    https://docs.greatexpectations.io/docs/guides/expectations/creating_custom_expectations/how_to_create_custom_column_map_expectations
+"""
+
+from typing import Dict, Optional
+
+from great_expectations.core.expectation_configuration import ExpectationConfiguration
+from great_expectations.exceptions.exceptions import (
+    InvalidExpectationConfigurationError,
+)
+from great_expectations.expectations.regex_based_column_map_expectation import (
+    ColumnMapRegexExpectation,
+    ColumnMapRegexMetricProvider,
+)
+
+
+# This class defines the Expectation itself
+class ExpectColumnValuesToMatchSomeRegex(ColumnMapRegexExpectation):
+    """TODO: Add a docstring here"""
+
+    # These values will be used to configure the metric created by your expectation
+    regex_snake_name = "vowel_name"
+    regex_camel_name = "VowelName"
+    regex = "regex pattern"
+
+    # These examples will be shown in the public gallery.
+    # They will also be executed as unit tests for your Expectation.
+    examples = []
+
+    def validate_configuration(self, configuration: Optional[ExpectationConfiguration]):
+        """
+        Validates that a configuration has been set, and sets a configuration if it has yet to be set. Ensures that
+        necessary configuration arguments have been provided for the validation of the expectation.
+
+        Args:
+            configuration (OPTIONAL[ExpectationConfiguration]): \
+                An optional Expectation Configuration entry that will be used to configure the expectation
+        Returns:
+            True if the configuration has been validated successfully. Otherwise, raises an exception
+        """
+
+        super().validate_configuration(configuration)
+        if configuration is None:
+            configuration = self.configuration
+
+        # # Check other things in configuration.kwargs and raise Exceptions if needed
+        # try:
+        #     assert (
+        #         ...
+        #     ), "message"
+        #     assert (
+        #         ...
+        #     ), "message"
+        # except AssertionError as e:
+        #     raise InvalidExpectationConfigurationError(str(e))
+
+        return True
+
+    # Here your regex is used to create a custom metric for this expectation
+    map_metric = ColumnMapRegexExpectation._register_metric(
```

Comment:
Has this pattern been arch reviewed?

Reply:
What is the purpose of this pattern?

Reply:
The goal is to enable a very streamlined development process for `ColumnMapRegexExpectations`, specifically by not needing to declare a separate `Metric` class.

Reply:
This pattern works in the sense of "executes (and makes the Expectation executable) without raising errors, failing tests, or producing immediately obvious side effects."

But I wrote it on a weekend without talking to anyone else. I'm not up to speed on the finer points of how our Expectation, Metric, and Renderer registries work, and I may be missing recent additions to our Code Standards.

So before committing to this pattern, I was hoping to get some other eyes on it.

(On the plus side, this is an internal method, so changing it wouldn't be a breaking API change.)

Reply:
Gotcha. That's really interesting. I don't think we've covered this in arch review, and it looks like we may not get a chance to for a few days. If we want to get this merged in more quickly, maybe we can get some thoughts from @alexsherstinsky ?

Reply:
Had a chance to review with @alexsherstinsky earlier today; after some experimentation and research post-discussion we're safe to move forward with an updated version of this pattern.